baybe/utils/device_utils.py

```diff
@@ -0,0 +1,130 @@
+"""Utilities for device management across BayBE."""
+
+import gc
+import os
+from collections.abc import Generator
+from contextlib import contextmanager
+from typing import Any
+
+import torch
+from gpytorch.settings import debug, fast_computations
+
+
+class _SingleDeviceMode:
+    """Internal context manager that forces operations to happen on a single device."""
```

Comment:
Notice that the way this class is written does not depend at all on the fact that its used to handle devices, i.e. its implementation is that of a generic settings handler --> rename?